src/DynamoCoreWpf/UI/GuidedTour/Step.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Windows.Controls.Primitives;
+using System.Windows.Media;
+
+namespace Dynamo.Wpf.UI.GuidedTour
+{
+    /// <summary>
+    /// This abstract class include several properties that will be used for childs like Survey, Welcome, Tooltip, ExitTour ....
+    /// </summary>
+    public abstract class Step
+    {
+        #region Public Properties
+        /// <summary>
+        /// The step content will contain the title and the popup content (included formatted text)
+        /// </summary>
+        public Content StepContent { get; set; }
+
+        /// <summary>
+        /// Step name, it just represent a step identifier
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// When the Step Width property is not provided the default value will be 480
+        /// </summary>
+        public double Width { get; set; } = 480;
+
+        /// <summary>
+        /// When the Step Height property is not provide the default value will be 190
+        /// </summary>
+        public double Height { get; set; } = 190;
+
+        /// <summary>
+        /// Represent a sequencial numeric value for each step, when is a multiflow guide this value can be repeated
+        /// </summary>
+        public int Sequence { get; set; } = 0;
+
+        /// <summary>
+        /// This static variable will be shared by all the steps and represent the total number of steps that the guide has.
+        /// </summary>
+        public static int TotalSteps { get; set; } = 8;
+
+        /// <summary>
+        /// This property contains the Host information like the host popup element or the popup position
+        /// </summary>
+        public HostInfo HostPopupInfo { get; set; }
+        public enum PointerDirection { TOP_RIGHT, TOP_LEFT, BOTTOM_RIGHT, BOTTOM_LEFT };
+
+        /// <summary>
+        /// This property describe which will be the pointing direction of the tooltip (if is a Welcome or Survey popup then is not used)
+        /// By default the tooltip pointer will be pointing to the left and will be located at the top
+        /// </summary>
+        protected PointerDirection TooltipPointerDirection { get; set; } = PointerDirection.TOP_LEFT;
```

Comment:
This should be moved to protected region?

Reply:
@QilongTang moved to the protected region.